src/ds/ds6/ds6-safety.cpp

```diff
@@ -184,4 +195,57 @@ namespace librealsense
 
         return results;
     }
+
+    void ds6_safety_sensor::set_safety_preset(int index, const rs2_safety_preset& sp) const
+    {
+        //// convert sp to vector of bytes
+        auto data_ptr = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&sp));
+
+        //check CRC
+        auto computed_crc32 = calc_crc32(data_ptr, sizeof(rs2_safety_preset));
+
+        // prepare vector of data
+        auto data_bytes = std::vector<uint8_t>(data_ptr, data_ptr + sizeof(sp));
+
+        // prepare header
+        rs2_safety_preset_header header = {1, 0xc0db, 405, computed_crc32 };
+        auto header_ptr = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&header));
+        auto header_bytes = std::vector<uint8_t>(header_ptr, header_ptr + sizeof(header));
+
+        //concat vector of data bytes to vector of header bytes
+        header_bytes.insert(header_bytes.end(), data_bytes.begin(), data_bytes.end());
```

Comment:
This is weird,
1.  `header_bytes` stores al the message?
2. We copy twice, once from `sp` to a byte vector, and again from the byte vector to another byte vector.
I think this could be optimized, right?

Reply:
done.
optimized the whole function. take a look on the new one